caliper_sender.py

```diff
@@ -134,39 +142,53 @@ def send_caliper_event():
 
 
 def get_caliper_event(event, event_type, event_action):
-    nav_path = document_path = chapter_path = ""
-    rsc = {}
-    if event.get('div_id'):
-        nav_path = event.get('div_id').split('/')
-        document_path = '/'.join(nav_path[:4]) + '/'
-        chapter_path = '/'.join(nav_path[:5]) + '/'
-        if len(nav_path) > 3:
-            rsc['document'] = nav_path[3]
-        if len(nav_path) > 4:
-            rsc['chapter'] = nav_path[4]
-        if len(nav_path) > 5:
-            rsc['page'] = nav_path[5]
+    # It's possible to not have chapters if it's a top level. This would result in -3 being 'published'
+    if not event.get('div_id'):
+        return None
+    nav_path = event.get('div_id').split('/')
```

Comment:
You mean as like a comment, okay I can put that in the code here.

Reply:
```suggestion
    # This will split paths like /srv/web2py/applications/runestone/books/thinkcspy/published/thinkcspy/GeneralIntro/Algorithms.html
    # See the unit test code in the tests directory for more examples
    nav_path = event.get('div_id').split('/')
```